src/test/resources/refaster/PreConditionsVerifierRecipes.java

```diff
@@ -0,0 +1,403 @@
+
+        /**
+         * Instantiates a new instance.
+         */
+        public NoUsesTypeWhenBeforeTemplateContainsPrimitiveAndAnotherTypeRecipe() {}
+
+        @Override
+        public String getDisplayName() {
+            return "Refaster template `PreConditionsVerifier.NoUsesTypeWhenBeforeTemplateContainsPrimitiveAndAnotherType`";
+        }
+
+        @Override
+        public String getDescription() {
+            return "Recipe created for the following Refaster template:\n```java\npublic static class NoUsesTypeWhenBeforeTemplateContainsPrimitiveAndAnotherType {\n    \n    @BeforeTemplate()\n    void before(int actual) {\n        System.out.println(actual);\n    }\n    \n    @BeforeTemplate()\n    void before(Map<?, ?> actual) {\n        System.out.println(actual);\n    }\n    \n    @AfterTemplate()\n    void after(Object actual) {\n        System.out.println(\"Changed: \" + actual);\n    }\n}\n```\n.";
+        }
+
+        @Override
+        public TreeVisitor<?, ExecutionContext> getVisitor() {
+            JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+                final JavaTemplate before = JavaTemplate
+                        .builder("System.out.println(#{actual:any(int)});")
+                        .build();
+                final JavaTemplate before0 = JavaTemplate
+                        .builder("System.out.println(#{actual:any(java.util.Map<?,?>)});")
+                        .build();
+                final JavaTemplate after = JavaTemplate
+                        .builder("System.out.println(\"Changed: \" + #{actual:any(java.lang.Object)});")
+                        .build();
+
+                @Override
+                public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
+                    JavaTemplate.Matcher matcher;
+                    if ((matcher = before.matcher(getCursor())).find()) {
+                        return embed(
+                                after.apply(getCursor(), elem.getCoordinates().replace(), matcher.parameter(0)),
+                                getCursor(),
+                                ctx,
+                                SHORTEN_NAMES
+                        );
+                    }
+                    if ((matcher = before0.matcher(getCursor())).find()) {
+                        maybeRemoveImport("java.util.Map");
+                        return embed(
+                                after.apply(getCursor(), elem.getCoordinates().replace(), matcher.parameter(0)),
+                                getCursor(),
+                                ctx,
+                                SHORTEN_NAMES
+                        );
+                    }
+                    return super.visitMethodInvocation(elem, ctx);
+                }
+
+            };
+            return Preconditions.check(
+                    Preconditions.and(
+                            new UsesType<>("java.util.Map", true),
```

Comment:
Name: **NoUsesTypeWhenBeforeTemplateContainsPrimitiveAndAnotherTypeRecipe**, thus:
- no `new UsesType<>("java.util.Map", true)`

Reply:
Turned into a code suggestion just below such that there's no ambiguity, as the wrapping `and` can also be removed.